### ISSUE TITLE
Feature request: respect "expand" query parameter when responding to create and update requests

### ISSUE THREAD

**Issue by Lex-2008** [pocketbase/pocketbase#160]
I like the possibility to expand "relation"-type fields - for example, when requesting a photo I can also get a name of album in which this photo is located: `client.Records.getOne("photos", "RECORD_ID", {expand: 'album'})`

I also like that in response to "update" request I get full record - I can use it to re-render relevant part of HTML. For example, when user first opens an editor, I run:
```
const record = await client.Records.getOne("photos", "RECORD_ID");
mainDiv.innerHTML=render(record);
```
And when they click "save" button:
```
const data = { ... };
const record = await client.Records.update("photos", "RECORD_ID", data);
mainDiv.innerHTML=render(record);
```

However, these two options don't play nicely together: if I add `{expand: 'album'}` as an extra argument to both "getOne" and "update" functions, then I see that js SDK adds query parameter in both cases, but pocketbase backend expands the relation only in first case. Probably it would be nice if it was used to to show the expanded data :-)

A workaround is, naturally, to make two requests:
```
const data = { ... };
await client.Records.update("photos", "RECORD_ID", data);
const record = await client.Records.getOne("photos", "RECORD_ID", {expand: 'album'});
mainDiv.innerHTML=render(record);
```
so it's not a very big issue, but would be nice to have :-)

**Comment by ganigeorgiev** [comment]
I understand your use case. I can't think of a reason why we cannot add it, so I'll consider it for the next minor bugfix/feature release, before starting working on the Dart SDK.

**Comment by ganigeorgiev** [comment]
Although this is not a bug, I released it with [v0.3.1](https://github.com/pocketbase/pocketbase/releases/tag/v0.3.1) since it is fairly small feature.

**Comment by ProfEibe** [comment]
is there a documentation how this works?
and is it possible to add a new child-relation without an id?

**Comment by ganigeorgiev** [comment]
> is there a documentation how this works?

It is available in the "API Preview" for each individual collection in the "Query parameters" table section. Currently it has the following description:

> Auto expand relations when returning the created record. Ex.: `?expand=rel1,rel2.subrel21.subrel22`
> Supports up to 6-levels depth nested relations expansion.
> The expanded relations will be appended to the record under the `@expand` property (eg. "@expand": {"rel1": {...}, ...}).
> Only the relations that the user has permissions to view will be expanded.

---

> and is it possible to add a new child-relation without an id?

I'm not sure that I understand the question. The expand is only for the return result after model creation. If you are asking for submitting together a base model + a relation together in a single request - this is not supported at the moment because it is ambiguous which collection API rule will apply.

**Comment by ProfEibe** [comment]
Alright thank you, this answered my question! 👍🏻
